modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/instance/SharedRocksDbInstance.java

```diff
@@ -0,0 +1,264 @@
+
+        resources.add(meta.columnFamily().handle());
+        resources.add(partitionCf.handle());
+        resources.add(gcQueueCf.handle());
+        resources.add(hashIndexCf.handle());
+        resources.addAll(sortedIndexCfs.values().stream()
+                .map(ColumnFamily::handle)
+                .collect(toList())
+        );
+
+        resources.add(db);
+        resources.add(flusher::stop);
+
+        try {
+            Collections.reverse(resources);
+
+            IgniteUtils.closeAll(resources);
+        } catch (Exception e) {
+            throw new StorageException("Failed to stop RocksDB storage: " + path, e);
+        }
+    }
+
+    /**
+     * Returns Column Family instance with the desired name. Creates it it it doesn't exist.
+     * Tracks every created index by its {@code indexId}.
+     */
+    public ColumnFamily getSortedIndexCfOnIndexCreate(byte[] cfName, int indexId) {
+        if (!busyLock.enterBusy()) {
+            throw new StorageClosedException();
+        }
+
+        try {
+            ColumnFamily[] result = {null};
+
+            sortedIndexIdsByCfName.compute(new ByteArray(cfName), (name, indexIds) -> {
+                ColumnFamily columnFamily = getOrCreateColumnFamily(cfName, name);
+
+                result[0] = columnFamily;
+
+                if (indexIds == null) {
+                    indexIds = new HashSet<>();
+                }
+
+                indexIds.add(indexId);
+
+                return indexIds;
+            });
+
+            return result[0];
+        } finally {
+            busyLock.leaveBusy();
+        }
+    }
+
+    /**
+     * Possibly drops the column family after destroying the index.
+     */
+    public void dropCfOnIndexDestroy(byte[] cfName, int indexId) {
+        if (!busyLock.enterBusy()) {
+            throw new StorageClosedException();
+        }
+
+        try {
+            sortedIndexIdsByCfName.compute(new ByteArray(cfName), (name, indexIds) -> {
+                if (indexIds == null) {
+                    return null;
+                }
+
+                indexIds.remove(indexId);
+
+                if (indexIds.isEmpty()) {
+                    indexIds = null;
+
+                    destroyColumnFamily(name);
+                }
+
+                return indexIds;
+            });
+        } finally {
+            busyLock.leaveBusy();
+        }
+    }
+
+    private ColumnFamily getOrCreateColumnFamily(byte[] cfName, ByteArray name) {
+        return sortedIndexCfs.computeIfAbsent(name, unused -> {
+            ColumnFamilyDescriptor cfDescriptor = new ColumnFamilyDescriptor(cfName, sortedIndexCfOptions(cfName));
+
+            ColumnFamily columnFamily;
+            try {
+                columnFamily = ColumnFamily.create(db, cfDescriptor);
+            } catch (RocksDBException e) {
+                throw new StorageException("Failed to create new RocksDB column family: " + new String(cfDescriptor.getName(), UTF_8), e);
```

Comment:
Maybe create a helper common method to get the name from **ColumnFamilyDescriptor**?

Reply:
I'll do it